kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -161,6 +175,19 @@ public LogSegment(
     this.version = version;
     this.deltas = deltas;
     this.checkpoints = checkpoints;
+    this.lastSeenChecksum =
+        lastSeenChecksum.filter(
+            file -> {
+              long checksumVersion = FileNames.checksumVersion(new Path(file.getPath()));
+              logger.debug(
+                  "Found checksum file for version {} in "
+                      + "LogSegment version {} and checkpoint version {}",
```

Comment:
IMO we could just do a logger.debug(this.toString) at the bottom of this constructor and print all the info?